### PR TITLE
UNR-537: Unreal GDK as a uplugin

### PR DESCRIPTION
**Pull request by danielimprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR restructures the GDK so that it can be built as a uplugin which simplifies the installation process.
#### Tests
This PR has been tested agains a branch of testsuite

#### Documentation
Additional PR with the docs will be created once merged.
#### Primary reviewers
@improbable-valentyn @joshuahuburn

### CODE REVIEW

File: SpatialGDK/Source/Public/SpatialGDKSettings.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/Public/SpatialGDKSettings.h]: Do we need this?

Comment (danielimprobable) [reply SpatialGDK/Source/Public/SpatialGDKSettings.h]: No that is probably a rebase conflict, @Vatyx this should be safe to remove right?

Comment (improbable-valentyn) [reply SpatialGDK/Source/Public/SpatialGDKSettings.h]: It's all commented out so would be _really_ strange if something breaks 😆

Comment (danielimprobable) [reply SpatialGDK/Source/Public/SpatialGDKSettings.h]: For sure it doesn't break, but I don't know if there was intent to readd it or to remove it

File: SpatialGDK/SpatialGDK.uplugin [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "EnabledByDefault" : true,
+	"CanContainContent": true,
+	"IsBetaVersion": false,
+	"Installed": true,
+	"Modules": [
+		{
+			"Name": "SpatialGDK",
+			"Type": "Runtime",
+			"LoadingPhase": "Default",
+            "WhitelistPlatforms": [ "Win64", "Linux" ]
```
Comment (m-samiec) [inline SpatialGDK/SpatialGDK.uplugin]: Add "XboxOne" and "PS4"?

Comment (danielimprobable) [reply SpatialGDK/SpatialGDK.uplugin]: I haven't been able to test those but I guess we could add them

File: SpatialGDK/Source/Private/Interop/SpatialTypebindingManager.cpp [hunk 3]
```diff
@@ -64,6 +64,7 @@ void USpatialTypebindingManager::CreateTypebindings()
 				}
 				else
 				{
+					RPCType = RPC_Count;
```
Comment (m-samiec) [inline SpatialGDK/Source/Private/Interop/SpatialTypebindingManager.cpp]: Drive-by?

File: SpatialGDKEditorToolbar/SpatialGDKEditorToolbar.uplugin [hunk 4]
```diff
@@ -17,7 +17,14 @@
 		{
 			"Name": "SpatialGDKEditorToolbar",
 			"Type": "Editor",
-			"LoadingPhase": "Default"
+			"LoadingPhase": "Default",
+            "WhitelistPlatforms": [ "Win64" ]
```
Comment (m-samiec) [inline SpatialGDKEditorToolbar/SpatialGDKEditorToolbar.uplugin]: Spacing off in this file?

File: SpatialGDK/Source/Private/Interop/SpatialTypebindingManager.cpp (hunk 3 above)
Comment (danielimprobable) [reply SpatialGDK/Source/Private/Interop/SpatialTypebindingManager.cpp]: linux compatibility fix, see https://github.com/spatialos/UnrealGDK/pull/338

File: SpatialGDK/Build/Scripts/BuildWorkerConfig.bat [hunk 5]
```diff
@@ -1,5 +1,5 @@
 @echo off
 
-pushd "%~dp0..\..\spatial"
+pushd "%~dp0..\..\..\..\..\..\spatial"
```
Comment (m-samiec) [inline SpatialGDK/Build/Scripts/BuildWorkerConfig.bat]: 😮

Comment (danielimprobable) [reply SpatialGDK/Build/Scripts/BuildWorkerConfig.bat]: Yep, this is a hack we have to have until we move to FPL

**Comment by danielimprobable** [comment]
Closed in favor of https://github.com/spatialos/UnrealGDK/pull/345
